openforcefield/tests/test_forcefield.py

```diff
@@ -643,8 +728,7 @@ def test_parameterize_ethanol_missing_torsion(self):
 </SMIRNOFF>
 ''')
         pdbfile = app.PDBFile(get_data_file_path('systems/test_systems/1_ethanol.pdb'))
-        molecules = []
-        molecules.append(Molecule.from_smiles('CCO'))
```

Comment:
In test where atom indexing is important, I've tried to remove uses of `Molecule.from_smiles`, since that does not guarantee a specific ordering of the atoms.

Reply:
> In tests where atom indexing is important, I've tried to remove uses of Molecule.from_smiles, since that does not guarantee a specific ordering of the atoms.

We can fix this! We can support tagged SMILES, such as `[C:1][C:2][O:3]`, and ensure that we create atoms with the specified ordering. This would allow the tests to be much more compact, and could have other uses in future.

I believe `cmiles` has methods to do this---we should consult with @ChayaSt .

Reply:
There have been requests for this functionality from several sources. I've opened an initial Issue with the spec for this: https://github.com/openforcefield/openforcefield/issues/448